spring-cloud-aws-dynamodb/src/test/java/io/awspring/cloud/dynamodb/DynamoDbTemplateIntegrationTest.java

```diff
@@ -0,0 +1,239 @@
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+@Testcontainers
+public class DynamoDbTemplateIntegrationTest {
+
+	private static DynamoDbTable dynamoDbTable;
+	private static DynamoDbTemplate dynamoDbTemplate;
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(DYNAMODB).withReuse(true);
+
+	@BeforeAll
+	public static void createTable() {
+		DynamoDbClient dynamoDbClient = DynamoDbClient.builder()
+				.endpointOverride(localstack.getEndpointOverride(DYNAMODB)).region(Region.of(localstack.getRegion()))
+				.credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("noop", "noop")))
+				.build();
+		DynamoDbEnhancedClient enhancedClient = DynamoDbEnhancedClient.builder().dynamoDbClient(dynamoDbClient).build();
+		dynamoDbTemplate = new DynamoDbTemplate(enhancedClient);
+		dynamoDbTable = DynamoDbEnhancedClient.builder().dynamoDbClient(dynamoDbClient).build().table("person_entity",
+				TableSchema.fromBean(PersonEntity.class));
+		dynamoDbTable.createTable();
+	}
+
+	@Test
+	void dynamoDbTemplate_saveAndRead_entitySuccessful() {
+		PersonEntity personEntity = PersonEntity.Builder.person().withName("foo").withLastName("bar")
+				.withUuid(UUID.randomUUID()).build();
+		assertThatCode(() -> dynamoDbTemplate.save(personEntity)).doesNotThrowAnyException();
+
+		PersonEntity personEntity1 = dynamoDbTemplate
+				.load(Key.builder().partitionValue(personEntity.getUuid().toString()).build(), PersonEntity.class);
+		assertThat(personEntity1).isEqualTo(personEntity);
+
+		cleanUp(personEntity.getUuid());
+	}
+
+	@Test
+	void dynamoDbTemplate_read_entitySuccessful_returnsNull() {
+		PersonEntity personEntity1 = dynamoDbTemplate
+				.load(Key.builder().partitionValue(UUID.randomUUID().toString()).build(), PersonEntity.class);
+		assertThat(personEntity1).isNull();
+	}
+
+	@Test
+	void dynamoDbTemplate_saveUpdateAndRead_entitySuccessful() {
+		PersonEntity personEntity = PersonEntity.Builder.person().withName("foo").withLastName("bar")
+				.withUuid(UUID.randomUUID()).build();
+		dynamoDbTemplate.save(personEntity);
+
+		personEntity.setLastName("xxx");
+		dynamoDbTemplate.update(personEntity);
+
+		PersonEntity personEntity1 = dynamoDbTemplate
+				.load(Key.builder().partitionValue(personEntity.getUuid().toString()).build(), PersonEntity.class);
+		assertThat(personEntity1).isEqualTo(personEntity);
+
+		// clean up
+		cleanUp(personEntity.getUuid());
+	}
+
+	@Test
+	void dynamoDbTemplate_saveAndDelete_entitySuccessful() {
+		PersonEntity personEntity = PersonEntity.Builder.person().withName("foo").withLastName("bar")
+				.withUuid(UUID.randomUUID()).build();
+		assertThatCode(() -> dynamoDbTemplate.save(personEntity)).doesNotThrowAnyException();
```

Comment:
I've started changing it but at some point I realised that maybe i am missing something - why is so much code in this test wrapped in `assertThatCode(...).doesNotThrowAnyException()`. I don't see like it adds anything to a test - if it throws, the test just fails, but perhaps i am missing something.